### PR TITLE
fix(source): validate descending ranges and guard timestamp counters

### PR DESCRIPTION
**Pull request by oritwoen**
## Summary
- reject descending and overflowing source ranges before processing to avoid underflow/overflow-driven incorrect workloads
- fix timestamp microseconds workload accounting to match actual processing (`+1 second` + `1000 ms variants` per second)
- add focused unit tests for descending range/date rejection, overflow guards, pre-epoch date rejection, and microseconds-mode counting

## Why
`RangeSource` and `TimestampSource` used inclusive-count arithmetic that could wrap on edge inputs (`end - start + 1`), and timestamp microseconds mode reported an incorrect total. This change makes source processing fail fast on invalid input and keeps progress/statistics consistent with real work.

## Verification
- `lsp_diagnostics` clean for modified files
- `cargo test`
- `cargo build`

### CODE REVIEW

File: src/source/timestamps.rs [hunk 1]
```diff
@@ -48,9 +63,32 @@ impl Source for TimestampSource {
...
+                )
+            })?;
         let total = if self.microseconds {
-            count * 1000
+            count.checked_mul(1001).ok_or_else(|| {
+                anyhow::anyhow!(
+                    "Timestamp workload overflow in microseconds mode for {} inputs",
+                    count
+                )
+            })?
```
[the remark this answers is not in the document]
Comment (oritwoen) [reply src/source/timestamps.rs]: Added the upfront bound check before the parallel loop in 4a151fc — rejects end timestamps that would wrap `ts * 1000 + ms`.
